testgrid/config/config.proto

```diff
@@ -2,6 +2,27 @@ syntax = "proto3";
 
 // Protocol buffer for configuring testgrid.k8s.io
 
+// Specifies the test name.
+message TestNameConfig {
+  // Specifies name elements to be selected from configuration values
+  message NameElement {
+    reserved 1, 3, 4;
+
+    // Configuration value to use.
+    // Valid choice are: 'Tests name', 'Commit',
+    // or any metadata key from finished.json, which is copied from your test suite.
```

Comment:
What about context and thread?

Also why not give a concrete example? No one is going to be able to follow these instructions. This is infinitely more clear:
```
+  test_name_config:
 +    name_elements:
 +    - target_config: Tests name
 +    - target_config: Context
 +    name_format: '%s [%s]'
```